src/Microsoft.DotNet.Interactive/KqlKernel.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Interactive.Commands;
+using Microsoft.DotNet.Interactive.Formatting;
+using static Microsoft.DotNet.Interactive.Formatting.PocketViewTags;
+
+namespace Microsoft.DotNet.Interactive
+{
+    /* This kernel is used as a placeholder for the MSKQL kernel in order to enable KQL language coloring
+* in the editor. Language grammars can only be defined for fixed kernel names, but MSKQL subkernels
+* are user-defined via the #!connect magic command. So, this kernel is specified in addition to the
+* user-defined kernel as a kind of "styling" kernel as well as to provide guidance and discoverability
+* for KQL features.
+*/
+    public class KQLKernel :
+        Kernel,
+        IKernelCommandHandler<SubmitCode>
+    {
+        private readonly HashSet<string> _kernelNameFilter;
+        public const string DefaultKernelName = "kql";
+
+        public KQLKernel() : base(DefaultKernelName)
+        {
+            _kernelNameFilter = new HashSet<string>
```

Comment:
This would be clearer and safer as `HashSet<KqlKernel>`

Reply:
I think the reason this was originally done is because the Microsoft.Dotnet.Interactive project doesn't have a reference to Microsoft.DotNet.Interactive.SqlServer.

Reply:
Oh right. That makes good sense.

Reply:
> I think the reason this was originally done is because the Microsoft.Dotnet.Interactive project doesn't have a reference to Microsoft.DotNet.Interactive.SqlServer.

correct :)